### ISSUE TITLE
Question: Disable stock fulfillment checking for this ingredient

### ISSUE THREAD

**Issue by kriddles** [grocy/grocy#548]
Disable stock fulfillment checking for this ingredient is a checkbox available in the recipe ingredients.
I see that this checkbox does properly update the visual saying there is "enough in stock" for the item.

Should this also trigger not consuming the ingredient?

I see the checkbox for only check single unit in stock does prevent consuming the ingredient.

Perhaps these two checkboxes should be merged?

**Comment by berrnd** [comment]
That's right - and also mentioned on the popup when consuming a recipe "ingredients marked with "check only if a single unit is in stock" will be ignored" that only those get ignored...

Maybe a separate option would be better (but we already have a lot of ingredient options)...

Or maybe a popup with all ingredients before consuming a recipe where you can uncheck those you want not to get consumed (like before adding missing ingredients to the shopping list)...

**Comment by kriddles** [comment]
> 
> Maybe a separate option would be better (but we already have a lot of ingredient options)...
> 

I'd prefer to not add more options as you mentioned there are already a lot.

If a recipe ingredient is "Disabled for stock check" then I would not know when trying to consume the recipe which ingredient would be causing the failure to consume. Due to this I assume the item would not be consumed.

A popup could provide more information and help with this especially since there is already a "are you sure check".

Perhaps group the items into categories like this?
"Consuming All Available Stock" (default Checked)
"Consuming" (default Checked)
"Not enough stock" (default UnChecked and helper text "Disabled for stock check")
"Ignored"(default UnChecked and helper text "Ignored due to "check only if single unit in stock"")

**Comment by berrnd** [comment]
Ok, then let's try to add such a popup with more information where products can be unchecked then, similar to what we already have when adding missing ingredients to the shopping list.

**Comment by alkuzman** [comment]
Would it be possible to do this issue in iterations? The way I see it, only one check `optional` is needed, which will consume the item if it is in stock. Otherwise, ignore it. No dialog needed for now.

I guess just rename `don't check stock fulfillment` into `optional` and make the necessary check during consumption.

I still don't understand the need for the other check 😕

**Comment by rugk** [comment]
See https://github.com/grocy/grocy/issues/2533 for a suggestion on the behavior of such optional ingredients:

> ### Alternative idea
> 
> A very cool solution for the issue at hand would be that, in case of such an "optional ingredient" in the recipe:
> * if it is tried to be consumed, show a list of all optional ones (and how much you actually have in stock), _and_ you can (de)select each ingredient
> * Recipe can obviously only be consumed if ingredients that are in stock have been checked.
> * All ingredients selected are then consumed…
> 
> This would circumvent the error, as no ingredients are tried to be consumed, which are not in stock.
